### PR TITLE
Test utility to start an anvil instance

### PR DESCRIPTION
**Pull request by neekolas**
### TL;DR

Added a utility package for starting and managing Anvil Ethereum test nodes in tests.

## Issues
https://github.com/xmtp/xmtpd/pull/630

### What changed?

- Created a new `anvil` package under `pkg/testutils` with functions to:
  - Find free network ports
  - Start Anvil instances with proper cleanup
  - Wait for Anvil to be ready before proceeding with tests
- Added comprehensive tests that verify:
  - Basic Anvil startup and connection
  - Concurrent operation of multiple Anvil instances

### How to test?

Run the tests in the new package:
```
go test -v ./pkg/testutils/anvil
```

The tests will verify that Anvil can be started, connected to, and that multiple instances can run concurrently.

### Why make this change?

This utility simplifies testing code that interacts with Ethereum by providing a consistent way to spin up local Anvil instances. It handles port allocation, startup synchronization, and proper cleanup, making blockchain-dependent tests more reliable and easier to write.

## Summary by CodeRabbit

- **New Features**
  - Introduced a testing utility that automates launching local test instances with dynamic port allocation and readiness verification.
  - Added a method to access the `RatesManager` instance within the `RatesAdmin` struct.
  - Added new metrics for tracking performance in payer operations, including node publishing duration and current nonce.
  - Introduced functions to retrieve absolute maximum and minimum payload sizes in the `GroupMessages` and `IdentityUpdates` contracts.
  
- **Tests**
  - Added comprehensive tests to verify both single and concurrent initialization of test instances, ensuring reliable startup and proper connection setup.
  - Updated tests to utilize a local Ethereum node for improved contract deployment and error handling.
  - Restructured test cases for `GroupMessages` and `IdentityUpdates` contracts, enhancing clarity and robustness.
  - Enhanced tests for the `RatesManager` contract to cover new functionalities and edge cases.
  - Removed deprecated functions and replaced them with updated utility functions for improved port allocation in test cases.

### CODE REVIEW

**Comment by neekolas** [comment]
* **#629** <a href="https://app.graphite.dev/github/pr/xmtp/xmtpd/629?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/xmtp/xmtpd/629?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: pkg/testutils/anvil/anvil.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+	"github.com/stretchr/testify/require"
+	"github.com/xmtp/xmtpd/pkg/blockchain"
+)
+
+func findFreePort() (int, error) {
+	ln, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return 0, err
+	}
```
Comment (mkysel) [inline pkg/testutils/anvil/anvil.go:21]: server_test has exactly the same function called `getNextOpenPort` can we unify them?

Comment (neekolas) [reply pkg/testutils/anvil/anvil.go:21]: Good catch. Will use the other one
